### PR TITLE
[16.0][MIG] excel_import_export, excel_import_export_demo: Migration to 16.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
@qrtl

### CODE REVIEW

**Review by kanda999 — APPROVED** [review]
Function review: it worked as expected in my environment.

**Comment by youring** [comment]
There are three same warnings when install the module:
```
2023-02-17 02:07:56,322 3666754 WARNING v16 odoo.api.create: The model odoo.addons.excel_import_export.models.xlsx_template is not overriding the create method in batch
2023-02-17 02:07:56,323 3666754 WARNING v16 odoo.api.create: The model odoo.addons.excel_import_export.models.xlsx_template is not overriding the create method in batch
2023-02-17 02:07:56,323 3666754 WARNING v16 odoo.api.create: The model odoo.addons.excel_import_export.models.xlsx_template is not overriding the create method in batch

**Comment by gaikaz** [comment]
@AungKoKoLin1997 
Could you please check those warnings youring mentioned?

**Comment by AungKoKoLin1997** [comment]
@youring @gaikaz 
Thanks for the review. You can see the update in the last [commit](https://github.com/OCA/server-tools/pull/2505/commits/43b5ad6e7bfe855d2928ab7acf294d818a574747)

**Comment by Dariosi** [comment]
@AungKoKoLin1997
This [commit ](https://github.com/OCA/server-tools/pull/2505/commits/43b5ad6e7bfe855d2928ab7acf294d818a574747 ) solves 1 out of 3 warning issues.
I guess the same should be done [here ](https://github.com/OCA/server-tools/pull/2505/files#diff-81f78a104b89cfbab95fe04902ac05aaceec2e38cf37c23495d0010dec5deceeR557-R560) and [here](https://github.com/OCA/server-tools/pull/2505/files#diff-81f78a104b89cfbab95fe04902ac05aaceec2e38cf37c23495d0010dec5deceeR605-R608)
(xlsx_template.py: 557, 605)

File: excel_import_export/controllers/report.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            excel = base64.decodebytes(excel)
+            if docids:
+                records = request.env[report.model].browse(docids)
+                if report.print_report_name and not len(records) > 1:
+                    report_name = safe_eval(
+                        report.print_report_name, {"object": records, "time": time}
+                    )
+                    # this is a bad idea, this sould only be .xlsx
+                    extension = report_name.split(".")[-1:].pop()
+                    report_name = f"{report_name}.{extension}"
```
Comment (gaikaz) [inline excel_import_export/controllers/report.py]: @AungKoKoLin1997 
It seems that you took the module while it was still in migration to Odoo 15 (https://github.com/OCA/server-tools/pull/2329).
Before it was fully done, there were some force-pushes that fixed some code. 
The forced commit: https://github.com/OCA/server-tools/pull/2329/commits/f94ba4fc5c3a15d979424cd043f49d2350e9df60#diff-db4160b3be01859e4c870cbbce8305e536fd4b5cad80ce3812b8e7a187aafccfR51-R56
The compare between force-pushed and commit before: https://github.com/OCA/server-tools/compare/88258a039d89c9050311a65c78cd7ae679c1463d..5f43f4d41911488b92b84aaf71b968e0b2ab13dd
Please also look what else changed.

While migrating, @Mantux11 made an error that switched the lines around. That started generating bad file names. Then he force-pushed to switch them back. 
It looks like you took the code before it was fixed.

TLDR:
```suggestion
                    # this is a bad idea, this should only be .xlsx
                    extension = report_name.split(".")[-1:].pop()
                    report_name = safe_eval(
                        report.print_report_name, {"object": records, "time": time}
                    )
                    report_name = f"{report_name}.{extension}"
```

**Review by gaikaz — CHANGES_REQUESTED** [review]
It seems that everything else should be OK.
We are testing the module for where we use it and apart from this one bug, it functionally works.

File: excel_import_export/static/src/js/report/action_manager_report.esm.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        );
+    }
+    if (onClose) {
+        onClose();
+    }
+}
+
+registry
+    .category("ir.actions.report handlers")
+    .add("xlsx_handler", async function (action, options, env) {
```
Comment (gaikaz) [inline excel_import_export/static/src/js/report/action_manager_report.esm.js]: Writing the last comment, found this too.

This is wrong, because it conflicts with another OCA module `report_xlsx`.
It also registers it's handler by the same name here:
https://github.com/OCA/reporting-engine/blob/16.0/report_xlsx/static/src/js/report/action_manager_report.esm.js#L8

If left like this, if you install both modules, only one handler (last installed) will be working and Odoo should be giving errors in the browser DevTools console.

```suggestion
    .add("excel_handler", async function (action, options, env) {
```

File: excel_import_export/controllers/report.py (hunk 1 above)
Comment (AungKoKoLin1997) [reply excel_import_export/controllers/report.py]: Yes. I worked on these modules while it was still in migration to 15.0. I added the commit for adjustment.

**Review by gaikaz — APPROVED** [review]
Code review + quick functional test.

**Comment by gaikaz** [comment]
@OCA/server-environment-maintainers 
Could we get an approve/merge from a maintainer please 🙏🏻

**Comment by gaikaz** [comment]
@AungKoKoLin1997 This is definitely all ready for review, right?
I already approved, as we depend on this module and AFAIK, it works as intended.

**Comment by AungKoKoLin1997** [comment]
@gaikaz Of course. This PR was ready for review. I don't know why still not get merged.

**Comment by gaikaz** [comment]
@OCA/tools-maintainers 
I suppose I pinged the wrong team before. 
Could we get a look at this 🙏🏻 Thank you
